hrm-service/service-tests/csam-reports.test.ts

```diff
@@ -0,0 +1,187 @@
+  Authorization: `Basic ${Buffer.from(process.env.API_KEY).toString('base64')}`,
+};
+
+const csamReports2DestroyQuery = {
+  where: {
+    accountSid: {
+      [Sequelize.Op.and]: [accountSid],
+    },
+  },
+};
+
+beforeAll(async () => {
+  await CSAMReport.destroy(csamReports2DestroyQuery);
+});
+
+afterAll(done => {
+  server.close(() => {
+    CSAMReport.destroy(csamReports2DestroyQuery).then(() => {
+      console.log('csam reports test cleaned up.');
+      done();
+    });
+  });
+});
+
+describe('/csamReports route', () => {
+  const route = `/v0/accounts/${accountSid}/csamReports`;
+
+  // const csamReportId = 'csam-report-id';
+
+  const expected = {
+    id: expect.anything(),
+    accountSid: accountSid,
+    csamReportId: 'csam-report-id',
+    twilioWorkerId: 'worker-sid',
+    contactId: null,
+    updatedAt: expect.toParseAsDate(),
+    createdAt: expect.toParseAsDate(),
+  };
+
+  describe('POST', () => {
+    test('should return 401', async () => {
+      const response = await request.post(route).send(csamReport1);
+
+      expect(response.status).toBe(401);
+      expect(response.body.error).toBe('Authorization failed');
+    });
+
+    test('should return 200', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport1);
+
+      expect(response.status).toBe(200);
+      expect(response.body).toStrictEqual(expected);
+
+      const reportFromDB = await CSAMReportController.getCSAMReport(response.body.id, accountSid);
+      /**
+       * Notes: console logging reportFromDB is not returning anything in console. I am thinking the 2 statements below are not being executed.
+       *
+       */
+      expect(reportFromDB).toBeDefined();
+      expect(reportFromDB.csamReportId).toEqual(expected.csamReportId);
+    });
+
+    test('unsure - when request is missing twilioWorkerId', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport2);
+      expect(response.status).toBe(200);
+      expect(response.body.twilioWorkerId).toEqual('');
+      /**
+       * Notes: This is just a passing test. I was expecting this test to have a status of 500 and returning an empty response body
+       *
+       */
+    });
+    test('unsure - when request is missing csamReportId', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport3);
+      expect(response.status).toBe(200);
+      expect(response.body.csamReportId).toEqual('');
+      /**
+       * Notes: This is just a passing test. I was expecting this test to have a status of 500 and returning an empty response body
+       *
+       */
+    });
+    test('unsure - scenario when contactId is part of response', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport4);
+      expect(response.status).toBe(500);
```

Comment:
This is most likely failing because the contact ID you specified (1234) probably doesn't have a contact for it in the DB. The database has built in referential integrity checks to prevent incorrect links like this being created - if you try to create a contact with a caseId, the caseId has to point at a real case or it will error, the same with csam report -> contact references.

This is a valid test - the endpoint SHOULD error when passed a non-existent contact id in it's payload (although it should probably be a 404 error, not a 500, but again, for another PR).

But there should be another test where a contact is created, and it's ID is used in the CSAM report, to check that this returns a 200 and updates the database correctly